### PR TITLE
Speed up ReadEventsAsyncPublishesEventsWithxIterators tests

### PR DESCRIPTION
**Pull request by christothes**
## Summary of Changes
The `ReadEventsAsyncPublishesEventsWithMultipleIterators` and `ReadEventsAsyncPublishesEventsWithOneIterator` tests have been fluctuating between sub-second runtime and 25+ seconds of runtime. This PR introduces a short `Task.Delay` call in the `PublishingTransportConsumerMock.ReceiveAsync` method, which reduces `Task` starvation in these test runs.

This change will trim up to a 50+ seconds of CI runtime per iteration of these two tests.

image
image

### CODE REVIEW

File: sdk/eventhub/Azure.Messaging.EventHubs/tests/Consumer/EventHubConsumerClientTests.cs [hunk 1]
```diff
@@ -2310,15 +2310,19 @@ private class PublishingTransportConsumerMock : TransportConsumer
...
                 var stopWatch = Stopwatch.StartNew();
                 PublishDelayCallback?.Invoke();
                 stopWatch.Stop();
 
                 if (((maximumWaitTime.HasValue) && (stopWatch.Elapsed >= maximumWaitTime)) || (PublishIndex >= EventsToPublish.Count))
                 {
-                    return Task.FromResult(Enumerable.Empty<EventData>());
+                    // Delay execution in this path to prevent a tight loop, starving other Tasks.
+
+                    await Task.Delay(100);
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/tests/Consumer/EventHubConsumerClientTests.cs]: Interesting.... this hadn't occurred to me at all.   Great catch!   

Should we consider `ConfigureAwait` here to further reduce contention?

**Review by jsquire — APPROVED** [review]
Thank you for this!   LGTM.   Please let us know when you're ready and we'll get this merged in.

File: sdk/eventhub/Azure.Messaging.EventHubs/tests/Consumer/EventHubConsumerClientTests.cs (hunk 1 above)
Comment (christothes) [reply sdk/eventhub/Azure.Messaging.EventHubs/tests/Consumer/EventHubConsumerClientTests.cs]: Yes, good catch on my catch 😄

Comment (christothes) [reply sdk/eventhub/Azure.Messaging.EventHubs/tests/Consumer/EventHubConsumerClientTests.cs]: Fixed, and ready to merge.

**Pull request merged**